perllib/FixMyStreet/Roles/Cobrand/SLWP.pm

```diff
@@ -316,9 +326,84 @@ sub garden_current_service_from_service_units {
     return $garden;
 }
 
+sub garden_container_data_extract {
+    my ($self, $data, $containers, $quantities, $schedules) = @_;
+    # Assume garden will only have one container data
+    my $garden_container = $containers->[0];
+    my $garden_bins = $quantities->{$containers->[0]};
+    if ($garden_container == CONTAINER_GARDEN_SACK) {
+        my $garden_cost = $self->garden_waste_renewal_sacks_cost_pa($schedules->{end_date}) / 100;
+        return ($garden_bins, 1, $garden_cost, $garden_container);
+    } else {
+        my $garden_cost = $self->garden_waste_renewal_cost_pa($schedules->{end_date}, $garden_bins) / 100;
+        return ($garden_bins, 0, $garden_cost, $garden_container);
+    }
+}
+
 # We don't have overdue renewals here
 sub waste_sub_overdue { 0 }
 
+# Same as full cost
+sub waste_get_pro_rata_cost {
+    my ($self, $bins, $end) = @_;
+    return $self->garden_waste_cost_pa($bins);
+}
+
+sub waste_garden_sub_params {
+    my ($self, $data, $type) = @_;
+    my $c = $self->{c};
+
+    my $service = $self->garden_current_subscription;
+    my $choice = $data->{container_choice} || '';
+    my $existing = $service ? $service->{garden_container} : undef;
+    my $container;
+    if ($choice eq 'sack') {
+        $container = CONTAINER_GARDEN_SACK;
+    } elsif ($choice eq 'bin140') {
+        $container = CONTAINER_GARDEN_BIN_140;
+    } elsif ($choice eq 'bin240') {
+        $container = CONTAINER_GARDEN_BIN;
+    } elsif ($choice) {
```

Comment:
You could remove the checks for `$choice eq 'bin240'` and `$choice` because everything defaults to CONTAINER_GARDEN_BIN in the final `else`.

Reply:
I could remove the bin240, though decided to leave it in to be clear, but can't remove $choice entirely because if choice exists it needs to take precedence over existing (which could be something else), if you see what I mean.